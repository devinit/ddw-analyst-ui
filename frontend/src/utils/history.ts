/* eslint-disable @typescript-eslint/naming-convention */
import axios, { AxiosResponse } from 'axios';
import * as localForage from 'localforage';
import { api, localForageKeys } from '.';
import { FrozenData } from '../components/SourceHistoryListItem/utils';
import { SavedQueryData } from '../components/DatasetHistoryCard/utils/types';

const SOURCE_HISTORY_BASEPATH = api.routes.FETCH_SOURCE_HISTORY;
const OPERATION_HISTORY_BASEPATH = api.routes.FETCH_DATASET_HISTORY;

export interface QueryResult<T = FrozenData> {
  count: number;
  next: number | null;
  previous: number | null;
  results: Array<T>;
}

interface FetchOptions {
  limit?: number;
  offset?: number;
}

export const fetchDataSourceHistory = async (
  id: number,
  options: FetchOptions = { limit: 10, offset: 0 },
): Promise<AxiosResponse<QueryResult>> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);
  const headers = {
    'Content-Type': 'application/json',
    Authorization: `token ${token}`,
  };

  return await axios(
    `${SOURCE_HISTORY_BASEPATH}${id}?limit=${options.limit || 10}&offset=${options.offset || 0}`,
    { headers },
  );
};

export const fetchFrozenData = async (id: number): Promise<AxiosResponse<FrozenData>> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);
  const headers = {
    'Content-Type': 'application/json',
    Authorization: `token ${token}`,
  };

  return await axios(`${api.routes.FROZEN_DATA}${id}`, { headers });
};

export const createFrozenData = async (data: FrozenData): Promise<AxiosResponse> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);
  const response = await axios.request({
    url: api.routes.FROZEN_DATA,
    method: 'post',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `token ${token}`,
    },
    data,
  });

  return response;
};

export const deleteFrozeData = async (id: number): Promise<AxiosResponse> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);
  const response = await axios.request({
    url: `${api.routes.FROZEN_DATA}${id}`,
    method: 'delete',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `token ${token}`,
    },
  });

  return response;
};

export const fetchOperationHistory = async (
  id: number,
  options: FetchOptions = { limit: 10, offset: 0 },
): Promise<AxiosResponse<QueryResult<SavedQueryData>>> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);
  const headers = {
    'Content-Type': 'application/json',
    Authorization: `token ${token}`,
  };

  return await axios(
    `${OPERATION_HISTORY_BASEPATH}${id}?limit=${options.limit || 10}&offset=${options.offset || 0}`,
    { headers },
  );
};

export const fetchSavedQueryData = async (id: number): Promise<AxiosResponse<SavedQueryData>> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);
  const headers = {
    'Content-Type': 'application/json',
    Authorization: `token ${token}`,
  };

  return await axios(`${api.routes.SINGLE_SAVED_QUERYSET}${id}`, { headers });
};

export const createSavedQueryData = async (data: SavedQueryData): Promise<AxiosResponse> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);
  const response = await axios.request({
    url: api.routes.SAVED_QUERYSETS,
    method: 'post',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `token ${token}`,
    },
    data,
  });

  return response;
};

export const deleteSavedQueryData = async (id: number): Promise<AxiosResponse> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);
  const response = await axios.request({
    url: `${api.routes.SINGLE_SAVED_QUERYSET}${id}`,
    method: 'delete',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `token ${token}`,
    },
  });

  return response;
};
