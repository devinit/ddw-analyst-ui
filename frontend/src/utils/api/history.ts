import axios, { AxiosResponse } from 'axios';
import * as localForage from 'localforage';
import { localForageKeys, api } from '..';
import { FrozenData } from '../../components/SourceHistoryListItem/utils';

const BASEPATH = api.routes.FETCH_SOURCE_HISTORY;

export interface ResultData {
  id: number;
  name: string;
  updated_on: string;
  is_draft: boolean;
  operation_query: string;
}

export interface QueryResult {
  data: Array<ResultData>;
}

interface FetchOptions {
  limit?: number;
  offset?: number;
}

export const fetchDataSourceHistory = async (
  id: number,
  options: FetchOptions = { limit: 10, offset: 0 },
): Promise<QueryResult> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);
  const headers = {
    'Content-Type': 'application/json',
    Authorization: `token ${token}`,
  };

  return await axios(
    `${BASEPATH}${id}?limit=${options.limit || 10}&offset=${options.offset || 0}`,
    { headers },
  );
};

export const createFrozenData = async (data: FrozenData): Promise<AxiosResponse> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);
  const response = await axios.request({
    url: api.routes.FROZEN_DATA,
    method: 'post',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `token ${token}`,
    },
    data,
  });

  return response;
};
