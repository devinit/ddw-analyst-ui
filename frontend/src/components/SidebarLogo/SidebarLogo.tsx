import { LocationDescriptor } from 'history';
import * as React from 'react';
import { Link } from 'react-router-dom';

interface LogoProps {
  url: LocationDescriptor;
  variation?: 'mini' | 'normal';
}

export const Logo: React.SFC<LogoProps> = props => {
  return (
    <Link to={ props.url } className={ `simple-text logo-${props.variation}` }>
      { props.children }
    </Link>
  );
};

Logo.defaultProps = {
  variation: 'normal'
};

Logo.displayName = 'Logo';

export class SidebarLogo extends React.Component {
  static Item = Logo;

  render() {
    return (
      <div className="logo">
        { this.renderContent() }
      </div>
    );
  }

  private renderContent(): React.ReactNode {
    return React.Children.map(this.props.children, child => {
      if (React.isValidElement(child) && child.type === Logo) {
        return child;
      }
    });
  }
}
