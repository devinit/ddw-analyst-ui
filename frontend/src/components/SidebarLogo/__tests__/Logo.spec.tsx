/**
 * @jest-environment jsdom
 */
import { createBrowserHistory } from 'history';
import * as React from 'react';
import { Router } from 'react-router';
import * as TestRenderer from 'react-test-renderer';
import { Logo } from '../SidebarLogo';

test('renders the correct url with the default props', () => {
  const renderer = TestRenderer.create(
    <Router history={createBrowserHistory()}>
      <Logo url="/path" />
    </Router>,
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders the mini variation when specified', () => {
  const renderer = TestRenderer.create(
    <Router history={createBrowserHistory()}>
      <Logo url="/link" variation="mini" />
    </Router>,
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders the normal variation when specified', () => {
  const renderer = TestRenderer.create(
    <Router history={createBrowserHistory()}>
      <Logo url="/link" variation="normal" />
    </Router>,
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});
