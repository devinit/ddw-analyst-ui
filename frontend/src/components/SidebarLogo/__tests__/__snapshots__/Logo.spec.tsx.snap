// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct url with the default props 1`] = `
<a
  className="simple-text logo-normal"
  href="/path"
  onClick={[Function]}
/>
`;

exports[`renders the mini variation when specified 1`] = `
<a
  className="simple-text logo-mini"
  href="/link"
  onClick={[Function]}
/>
`;

exports[`renders the normal variation when specified 1`] = `
<a
  className="simple-text logo-normal"
  href="/link"
  onClick={[Function]}
/>
`;
